From e8995d4e1d5cf984cf10327c59808976425b2f9c Mon Sep 17 00:00:00 2001
From: Carlos Garcia Campos <cgarcia@igalia.com>
Date: Fri, 20 Apr 2018 12:14:55 +0200
Subject: [PATCH] soup-socket: fix critical warning when the peer-certificate
 changes after disconnect

(process:15586): GLib-GIO-CRITICAL **: 12:14:39.627: g_tls_connection_get_peer_certificate_errors: assertion 'G_IS_TLS_CONNECTION (conn)' failed

We should disconnect the signal on disconnect_internal before clearing
the connection object.
---
 libsoup/soup-socket.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libsoup/soup-socket.c b/libsoup/soup-socket.c
index 5363dc72..311c627c 100644
--- a/libsoup/soup-socket.c
+++ b/libsoup/soup-socket.c
@@ -188,6 +188,7 @@ disconnect_internal (SoupSocket *sock, gboolean close)
 	g_clear_object (&priv->gsock);
 	if (priv->conn && close) {
 		g_io_stream_close (priv->conn, NULL, NULL);
+		g_signal_handlers_disconnect_by_data (priv->conn, sock);
 		g_clear_object (&priv->conn);
 	}
 
-- 
2.19.1

